cli/internal/cmd/terminate.go

```diff
@@ -45,13 +44,27 @@ func runTerminate(cmd *cobra.Command, args []string) error {
 
 func terminate(cmd *cobra.Command, terminator cloudTerminator, fileHandler file.Handler, spinner spinnerInterf,
 ) error {
-	var idFile clusterid.File
-	if err := fileHandler.ReadJSON(constants.ClusterIDsFileName, &idFile); err != nil {
+	foundTerraformData, err := checkForTerraformData(fileHandler)
+	if err != nil {
 		return err
 	}
 
+	if !foundTerraformData {
+		return fmt.Errorf("did not find a cluster to terminate")
+	}
```

Comment:
I don't think we should check for terraform files for 2 reasons:

1. There might be resources we want to remove that are not dependent on terraform (e.g. libvirt container)
2. Terraform won't perform any actions if there are no terraform files in the workspace, meaning we don't need to check ourselves

Reply:
Maybe other people have an opinion on this?
IMO if there is nothing to do, `terminate` should not error

Reply:
The current behaviour was to error so I just switch to an error for consistency.
Can switch to `fmt.Println` what I initially had. That's fine, too.

Reply:
> IMO if there is nothing to do, terminate should not error. 

I agree. Our CLI should be idempotent for resource termination. This means: If there are dangling resources, try cleaning them up. If everything is cleaned up, remove the relevant state files. However, if cleanup partially succeeded, allow a user to intervene and make it safe to rerun `constellation terminate` any number of times (until it returns with exit code 0 and the state file if removed).
If a user (or automated system) reruns `constellation terminate` when there is nothing to terminate, we can print a message to indicate that there is no cleanup to be done but should indicate success using an exit code of 0.

This makes it very easy to integrate the CLI into other tools as we can ensure that there is no state left over from a previous constellation before creating a new one.

Reply:
Definitely agree with not returning an error personally. Was really just for consistency with the previous behavior. Will change that.

I also like the idea of printing when there is nothing to do, hence I went for it. I think Daniel's point really was if we want to do that based on checking if certain files exist. However, I am not sure how else we can do that without major refactoring and actually checking Terraform and/or Docker to see if there is something to clean up.